tensorboard/plugins/pr_curve/summary_test.py

```diff
@@ -22,302 +22,241 @@
-    multiplexer.AddRunsFromDirectory(self.logdir)
-    multiplexer.Reload()
-    return multiplexer
-
-  def validateTensorEvent(self, expected_step, expected_value, tensor_event):
-    """Checks that the values stored within a tensor are correct.
-
-    Args:
-      expected_step: The expected step.
-      tensor_event: A TensorEvent named tuple.
-      expected_value: A nested python list of expected float32 values.
+  def pb_via_op(self, summary_op, feed_dict=None):
+    with tf.Session() as sess:
+      actual_pbtxt = sess.run(summary_op, feed_dict=feed_dict or {})
+    actual_proto = tf.Summary()
+    actual_proto.ParseFromString(actual_pbtxt)
+    return actual_proto
+
+  def normalize_summary_pb(self, pb):
+    """Pass `pb`'s `TensorProto` through a marshalling roundtrip.
+    `TensorProto`s can be equal in value even if they are not identical
+    in representation, because data can be stored in either the
+    `tensor_content` field or the `${dtype}_value` field. This
+    normalization ensures a canonical form, and should be used before
+    comparing two `Summary`s for equality.
+    """
+    result = tf.Summary()
+    result.MergeFrom(pb)
+    for value in result.value:
+      if value.HasField('tensor'):
+        new_tensor = tf.make_tensor_proto(tf.make_ndarray(value.tensor))
+        value.ClearField('tensor')
+        value.tensor.MergeFrom(new_tensor)
+    return result
+
+  def compute_and_check_summary_pb(self,
+                                   name,
+                                   labels,
+                                   predictions,
+                                   num_thresholds,
+                                   weights=None,
+                                   display_name=None,
+                                   description=None,
+                                   feed_dict=None):
+    """Use both `op` and `pb` to get a summary, asserting equality.
+    Returns:
+      a `Summary` protocol buffer
     """
-    self.assertEqual(expected_step, tensor_event.step)
-    tensor_nd_array = tf.make_ndarray(tensor_event.tensor_proto)
+    labels_tensor = tf.constant(labels)
+    predictions_tensor = tf.constant(predictions)
+    weights_tensor = None if weights is None else tf.constant(weights)
+    op = summary.op(
+        tag=name,
+        labels=labels_tensor,
+        predictions=predictions_tensor,
+        num_thresholds=num_thresholds,
+        weights=weights_tensor,
+        display_name=display_name,
+        description=description)
+    pb = self.normalize_summary_pb(summary.pb(
+        tag=name,
+        labels=labels,
+        predictions=predictions,
+        num_thresholds=num_thresholds,
+        weights=weights,
+        display_name=display_name,
+        description=description))
+    pb_via_op = self.normalize_summary_pb(
+        self.pb_via_op(op, feed_dict=feed_dict))
+    self.assertProtoEquals(pb, pb_via_op)
+    return pb
+
+  def verify_float_arrays_are_equal(self, expected, gotten):
```

Comment:
nit: Can we rename `gotten` to the more conventional term `actual`?

Reply:
Done.